### PR TITLE
Fix LAM inline post and long run time RT tests

### LINKED ISSUE #504: regional_quilt post files show run-to-run variability
## Description

During the commit of PR #475, the GRB files for regional_quilt failed to reproduce. 

### To Reproduce:

Create a baseline for the reqional_quilt test using the hash corresponding to PR #475. Attempt to verify the test against that new baseline.

## Additional context

Verification of baselines failed for either the NATLEV.GrbF00 and NATLEV.GrbF24 or the PRSLEV.GrbF00 and PRSLEV.GrbF24.
Additional information can be found in the comment thread [here](https://github.com/ufs-community/ufs-weather-model/pull/475#issuecomment-812054909)

### LINKED ISSUE #519: Changes in LAM configuration
## Description
LAM configuration has been changed. It is affecting two files:  suite_FV3_GFS_v15_thompson_mynn_lam3km.xml and input.nml

## Solution
Make change in  suite_FV3_GFS_v15_thompson_mynn_lam3km.xml SDF:
Use mynnsfc_wrapper instead of sfc_diff

In input.nml set:
     do_mynnsfclay = .true.

### LINKED ISSUE #525: Regression test `fv3_gfs_v16_RRTMGP_2thrd` overwrites the baseline created by `fv3_gfs_v16_RRTMGP`
## Description

The following line in `rt.conf` means that the test `fv3_gfs_v16_RRTMGP_2thrd` unnecessarily/mistakenly overwrites the baseline created by `fv3_gfs_v16_RRTMGP` - this may also cause problems if both tests finish at the same time.
```
RUN     | fv3_gfs_v16_RRTMGP_2thrd                                                                                                |                                         | fv3 |
```

### To Reproduce:

Look at the current `tests/rt.conf` and at `tests/tests/fv3_gfs_v16_RRTMGP` and `tests/tests/fv3_gfs_v16_RRTMGP_2thrd`

## Additional context

The threading test should only be run when verifying against an existing baseline (created with the non-threaded test).

## Solution

Remove `fv3` from the last column in the above line in `rt.conf`.

### PR DESCRIPTION
**Pull request by junwang-noaa**
# PR Checklist

- [x ] Ths PR is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR. Please consult the ufs-weather-model [wiki](https://github.com/ufs-community/ufs-weather-model/wiki/Making-code-changes-in-the-UFS-weather-model-and-its-subcomponents) if you are unsure how to do this.

- [x ] This PR has been tested using a branch which is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR

- [x ] An Issue describing the work contained in this PR has been created either in the subcomponent(s) or in the ufs-weather-model. The Issue should be created in the repository that is most relevant to the changes in contained in the PR. The Issue and the dependent sub-component PR 
are specified below.

- [ ] If new or updated input data is required by this PR, it is clearly stated in the text of the PR.
No input data change

## Instructions: All subsequent sections of text should be filled in as appropriate.

The information provided below allows the code managers to understand the changes relevant to this PR, whether those changes are in the ufs-weather-model repository or in a subcomponent repository. Ufs-weather-model code managers will use the information provided to add any applicable labels, assign reviewers and place it in the Commit Queue. Once the PR is in the Commit Queue, it is the PR owner's responsiblity to keep the PR up-to-date with the develop branch of ufs-weather-model. 

## Description

This PR will do the following:
1) Fix the LAM inline post reproducibility issue due to uninitialized variables. 
2) More tasks have been added in several RT tests that take long run time. Those tests were reported reach wall clock time on some platforms.   
3) Update UPP lib from 10.0.4 to 10.0.5. UPP lib10.0.5 has threading issue fixed.
4) Remove the baseline creation in fv3_gfs_v16_RRTMGP_2thrd test
5) change LAM configuration in LAM3km suite file and namelist

### Issue(s) addressed

Link the issues to be closed with this PR, whether in this repository, or in another repository.
(Remember, issues must always be created before starting work on a PR branch!) 
- fixes [#504](https://github.com/ufs-community/ufs-weather-model/issues/504)
- fixes #525 
- fixes #519

## Testing

Testing has been done on hera and orion.

-  ww3 tasks are increased from 42 to 82 in fv3 wave tests (fv3_gfdlmprad and fv3_gfdlmprad_atmwav) run time is changed from 700-800s to 350-400s
- forecast length fhmax is reduced from 6 to 3, fv3 write nemsio c768 with wave run time change: 710s ->530s
- total tasks in regional tests are increased from 24->80, run time change: 550->250s, results are also changed

How were these changes tested? What compilers / HPCs was it tested with? Are the changes covered by regression tests? (If not, why? Do new tests need to be added?) Have regression tests and unit tests (utests) been run? On which platforms and with which compilers? (Note that unit tests can only be run on tier-1 platforms)

- [x ] hera.intel
- [x ] hera.gnu
- [x ] orion.intel
- [x ] cheyenne.intel 
- [x ] cheyenne.gnu
- [x ] gaea.intel 
- [x ] jet.intel
- [x ] wcoss_cray
- [x ] wcoss_dell_p3

## Dependencies

If testing this branch requires non-default branches in other repositories, list them. Those branches should have matching names (ideally).

Do PRs in upstream repositories need to be merged first?
If so add the "waiting for other repos" label and list the upstream PRs
- waiting on [noaa-emc/fv3atm PR#272](https://github.com/NOAA-EMC/fv3atm/pull/272)

### CODE REVIEW

**Review by climbfuji — COMMENTED** [review]
Looks good to me. The fix for the LAM post is in the fv3atm PR, correct? Everything in this ufs-weather-models PR is about reducing the runtime of the regression tests?

**Comment by junwang-noaa** [comment]
@climbfuji Yes, the lam post fix is in FV3 PR, POST code manager reviewed the code changes. The ufs-weather-models PR is to reduce the run time of some long RT test.

**Review by DusanJovic-NOAA — APPROVED** [review]

**Comment by junwang-noaa** [comment]
When the module file update is committed, I will update upp lib to 10.0.5.

**Comment by BrianCurtis-NOAA** [comment]
Machine: gaea
Compiler: intel
Job: BL
Repo location: /lustre/f2/pdata/ncep/emc.nemspara/autort/pr/610295225/20210413023016/ufs-weather-model
Please manually delete: /lustre/f2/scratch/emc.nemspara/FV3_RT/rt_7367
Repo location: /lustre/f2/pdata/ncep/emc.nemspara/autort/pr/610295225/20210413033956/ufs-weather-model
Please manually delete: /lustre/f2/scratch/emc.nemspara/FV3_RT/rt_9828
Test fv3_regional_quilt 023 failed in check_result failed 
Test fv3_regional_quilt 023 failed in run_test failed 
Please make changes and add the following label back:
gaea-intel-BL

**Comment by BrianCurtis-NOAA** [comment]
Machine: orion
Compiler: intel
Job: BL
Repo location: /work/noaa/nems/emc.nemspara/autort/pr/610295225/20210412224511/ufs-weather-model
Please manually delete: /work/noaa/stmp/bcurtis/stmp/bcurtis/FV3_RT/rt_375545
Test fv3_regional_quilt 023 failed in run_test failed 
Please make changes and add the following label back:
orion-intel-BL

File: tests/rt.conf [hunk 1]
```diff
@@ -5,8 +5,8 @@
 COMPILE | APP=ATM SUITES=FV3_GFS_2017                                                                                             |                                         | fv3 |
 
 RUN     | fv3_control                                                                                                             |                                         | fv3 |
-RUN     | fv3_decomp                                                                                                              | - jet.intel                             |     |
-RUN     | fv3_2threads                                                                                                            |                                         |     |
+RUN     | fv3_decomp                                                                                                              | - jet.intel                             |     | fv3_control
+RUN     | fv3_2threads                                                                                                            |                                         |     | fv3_control
```
Comment (DusanJovic-NOAA) [inline tests/rt.conf]: Why do these two tests (fv3_decomp and fv3_2threads) now depend on fv3_control?

File: tests/rt.conf [hunk 2]
```diff
@@ -80,7 +80,7 @@ RUN     | fv3_gfs_v16_restart
 RUN     | fv3_gfs_v16_stochy                                                                                                      |                                         | fv3 |
 RUN     | fv3_gfs_v16_RRTMGP                                                                                                      |                                         | fv3 |
 RUN     | fv3_gfs_v16_RRTMGP_c192L127                                                                                             |                                         | fv3 |
-RUN     | fv3_gfs_v16_RRTMGP_2thrd                                                                                                |                                         | fv3 |
+RUN     | fv3_gfs_v16_RRTMGP_2thrd                                                                                                |                                         | fv3 | fv3_gfs_v16_RRTMGP
```
Comment (DusanJovic-NOAA) [inline tests/rt.conf]: This test (fv3_gfs_v16_RRTMGP_2thrd) should not depend on fv3_gfs_v16_RRTMGP.

Comment (climbfuji) [reply tests/rt.conf]: It should also have the `fv3` entry removed, see https://github.com/ufs-community/ufs-weather-model/issues/525

Comment (junwang-noaa) [reply tests/rt.conf]: I tried to fix the issue [#525](https://github.com/ufs-community/ufs-weather-model/issues/525).  Just realized the threading test is actually compared with baseline control, not the control running with threading. I will remove the fv3 in fv3_gfs_v16_RRTMGP_2thrd instead

**Comment by junwang-noaa** [comment]
Fixed

**Comment by climbfuji** [comment]
> Fixed
> […](#)

Thanks! I updated your PR description so that merging it will fix the issue I mentioned.

**Comment by BrianCurtis-NOAA** [comment]
Cheyenne AutoRT should be working too

**Comment by BrianCurtis-NOAA** [comment]
Machine: cheyenne
Compiler: gnu
Job: BL
Repo location: /glade/work/briancurtis/git/BrianCurtis-NOAA/ufs-weather-model/tests/auto/pr/610295225/20210419094510/ufs-weather-model
Please manually delete: /glade/scratch/briancurtis/FV3_RT/rt_66895
Repo location: /glade/work/briancurtis/git/BrianCurtis-NOAA/ufs-weather-model/tests/auto/pr/610295225/20210419105053/ufs-weather-model
Please manually delete: /glade/scratch/briancurtis/FV3_RT/rt_23761
Test fv3_gfs_v15p2_debug 022 failed in check_result failed 
Test fv3_gfs_v15p2_debug 022 failed in run_test failed 
Test fv3_esg_HAFS_v0_hwrf_thompson_debug 027 failed in check_result failed 
Test fv3_esg_HAFS_v0_hwrf_thompson_debug 027 failed in run_test failed 
Test fv3_regional_control_debug 017 failed in check_result failed 
Test fv3_regional_control_debug 017 failed in run_test failed 
Test fv3_gfs_v16_RRTMGP_debug 024 failed in check_result failed 
Test fv3_gfs_v16_RRTMGP_debug 024 failed in run_test failed 
Test fv3_esg_HAFS_v0_hwrf_thompson 012 failed in check_result failed 
Test fv3_esg_HAFS_v0_hwrf_thompson 012 failed in run_test failed 
Test fv3_gfs_v16_debug 023 failed in check_result failed 
Test fv3_gfs_v16_debug 023 failed in run_test failed 
Test fv3_HAFS_v0_hwrf_thompson_debug 026 failed in check_result failed 
Test fv3_HAFS_v0_hwrf_thompson_debug 026 failed in run_test failed 
Test fv3_gfsv16_ugwpv1_debug 028 failed in check_result failed 
Test fv3_gfsv16_ugwpv1_debug 028 failed in run_test failed 
Test fv3_control_debug 016 failed in check_result failed 
Test fv3_control_debug 016 failed in run_test failed 
Test fv3_gfs_v16_ras_debug 029 failed in check_result failed 
Test fv3_gfs_v16_ras_debug 029 failed in run_test failed 
Test fv3_gfs_v16_flake 005 failed in check_result failed 
Test fv3_gfs_v16_flake 005 failed in run_test failed 
Test fv3_thompson_debug 020 failed in check_result failed 
Test fv3_thompson_debug 020 failed in run_test failed 
Test fv3_rrfs_v1beta_debug 018 failed in check_result failed 
Test fv3_rrfs_v1beta_debug 018 failed in run_test failed 
Test fv3_thompson_no_aero_debug 021 failed in check_result failed 
Test fv3_thompson_no_aero_debug 021 failed in run_test failed 
Test fv3_gsd_debug 019 failed in check_result failed 
Test fv3_gsd_debug 019 failed in run_test failed 
Test fv3_gfdlmp 001 failed in check_result failed 
Test fv3_gfdlmp 001 failed in run_test failed 
Test fv3_gfs_v16_stochy 004 failed in check_result failed 
Test fv3_gfs_v16_stochy 004 failed in run_test failed 
Test fv3_gfs_v16 002 failed in check_result failed 
Test fv3_gfs_v16 002 failed in run_test failed 
Test fv3_thompson_no_aero 009 failed in check_result failed 
Test fv3_thompson_no_aero 009 failed in run_test failed 
Test fv3_thompson 008 failed in check_result failed 
Test fv3_thompson 008 failed in run_test failed 
Test fv3_gfs_v16_RRTMGP 006 failed in check_result failed 
Test fv3_gfs_v16_RRTMGP 006 failed in run_test failed 
Test fv3_rrfs_v1beta 010 failed in check_result failed 
Test fv3_rrfs_v1beta 010 failed in run_test failed 
Test fv3_gfs_v16_ras 015 failed in check_result failed 
Test fv3_gfs_v16_ras 015 failed in run_test failed 
Test fv3_gsd 007 failed in check_result failed 
Test fv3_gsd 007 failed in run_test failed 
Test fv3_HAFS_v0_hwrf_thompson 011 failed in check_result failed 
Test fv3_HAFS_v0_hwrf_thompson 011 failed in run_test failed 
Test fv3_gfsv16_ugwpv1 013 failed in check_result failed 
Test fv3_gfsv16_ugwpv1 013 failed in run_test failed 
Test fv3_multigases 025 failed in check_result failed 
Test fv3_multigases 025 failed in run_test failed 
Test fv3_gfsv16_ugwpv1_warmstart 014 failed in check_result failed 
Test fv3_gfsv16_ugwpv1_warmstart 014 failed in run_test failed 
Please make changes and add the following label back:
cheyenne-gnu-BL

**Comment by BrianCurtis-NOAA** [comment]
Baseline Cheyenne-GNU were created successfully. The move failed, but that has been fixed and I will be manually moving the Cheyenne BL's to the correct place and running the RT's on them

**Comment by junwang-noaa** [comment]
CI test passed.

**Review by RatkoVasic-NOAA — APPROVED** [review]

**Review by DeniseWorthen — APPROVED** [review]

**Review by climbfuji — APPROVED** [review]

**Pull request merged**